File: week1/src/basic_simple_listener.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <std_msgs/Int64.h>

void chatterCallback(const std_msgs::Int64& num)
{
  ROS_INFO("subscribe: %d", num.data);
}

int main(int argc, char** argv)
{
  ros::init(argc, argv, "basic_simple_listener");
  ros::NodeHandle nh;
  ros::Subscriber sub = nh.subscribe("chatter", 10, chatterCallback);

  ros::spin();
  return 0;
}

File: src/week1/script/basic_simple_listener.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import String
from std_msgs.msg import Int64

def callback(data):
    rospy.loginfo("subscribe:%d",data.data)

def listener():
    rospy.init_node('listener', anonymous=True)
    rospy.Subscriber("chatter", Int64, callback)
    rospy.spin()

if __name__ == '__main__':
    listener()